src/hooks/web3/useJsonRpcUrlMap.tsx

```diff
@@ -30,3 +15,36 @@ export default function useJsonRpcUrlMap(): Record<SupportedChainId, string[]> {
   invariant(jsonRpcUrlMap, 'useJsonRpcUrlMap used without initializing the context')
   return jsonRpcUrlMap
 }
+
+function toJsonRpcMap(connectionMap?: JsonRpcConnectionMap): Record<SupportedChainId, (JsonRpcProvider | string)[]> {
+  return ALL_SUPPORTED_CHAIN_IDS.reduce((map, chainId) => {
+    const value = connectionMap?.[chainId]
+    const connections = (Array.isArray(value) ? value : [value])
+      .filter((value): value is string | JsonRpcProvider => Boolean(value))
+      .concat(...JSON_RPC_FALLBACK_ENDPOINTS[chainId])
+    return {
+      ...map,
+      [chainId]: connections,
+    }
+  }, {} as Record<SupportedChainId, (JsonRpcProvider | string)[]>)
+}
+
+export function toJsonRpcConnectionMap(
+  connectionMap?: JsonRpcConnectionMap
+): Record<SupportedChainId, [JsonRpcProvider]> {
+  return Object.entries(toJsonRpcMap(connectionMap)).reduce((map, [chainId, connections]) => {
+    const connection = connections[0]
+    const provider = JsonRpcProvider.isProvider(connection) ? connection : new StaticJsonRpcProvider(connection)
```

Comment:
This must be provided with a default chainId, or it will immediately send a JSON-RPC request to determine the network.